crawl-ref/source/ability.cc

```diff
@@ -2882,21 +2938,21 @@ bool activate_talent(const talent& tal, dist *target)
             // Ephemeral Shield activates on any invocation with a cost,
             // even if that's just a cooldown or small amounts of HP.
             // No rapidly wall-jumping or renaming your ancestor, alas.
-            if (is_religious_ability(abil.ability)
-                && (abil.piety_cost || (abil.flags & abflag::exhaustion)
-                    || (abil.flags & abflag::max_hp_drain)
-                    || (abil.ability == ABIL_ZIN_RECITE)
-                    || (abil.flags & abflag::card) || (abil.flags & abflag::gold)
-                    || (abil.flags & abflag::sacrifice)
-                    || (abil.flags & abflag::torment)
-                    || (abil.flags & abflag::injury) || abil.get_hp_cost() > 0
-                    || abil.get_mp_cost() > 0)
+            if (_not_free_religious_ability(abil.ability)
                 && you.has_mutation(MUT_EPHEMERAL_SHIELD))
             {
                 you.set_duration(DUR_EPHEMERAL_SHIELD, random_range(3, 5));
                 you.redraw_armour_class = true;
             }
 
+            if (_not_free_religious_ability(abil.ability)
+                && you.unrand_equipped(UNRAND_DRAGONMASK)
+                && there_are_monsters_nearby(true, false, false))
```

Comment:
`there_are_monsters_nearby(true, true, false)` , please. This doesn't strictly change anything because the mask has sinv already, but we do want visibility-to-the-player to be enforced here in case anyone ever does a drive-by-change of the unrand's artprops in the future.